cloud/gcp/main.tf

```diff
@@ -0,0 +1,32 @@
+provider "google" {
+ alias = "impersonation"
```

Comment:
question (blocking): what does "impersonation" signify here?

Reply:
It's best practice to have terraform run with a service account. However, key generation for service account is disabled for .org projects. To bypass that, terraform 1st runs as 'my-user' (ktoor@google.com), makes a token request to terraform service account to 'impersonate' as that service account. Once that token is received, you can create another provider to impersonate as that service account thus making all further requests using that service account.

Reply:
impersonation keyword implies that the initial provider is just going to be used for impersonation purposes.

Reply:
Ahh that is super helpful, thank you. Would you please add a code comment capturing that?